static/app/components/core/inspector.tsx

```diff
@@ -0,0 +1,671 @@
+import {Fragment, useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {createPortal} from 'react-dom';
+import {usePopper} from 'react-popper';
+import type {Theme} from '@emotion/react';
+import {css} from '@emotion/react';
+import color from 'color';
+
+import {addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {Tag} from 'sentry/components/core/badge/tag';
+import {Flex, Stack} from 'sentry/components/core/layout';
+import {Separator} from 'sentry/components/core/separator';
+import {Text} from 'sentry/components/core/text';
+import {Overlay} from 'sentry/components/overlay';
+import {
+  ProfilingContextMenu,
+  ProfilingContextMenuGroup,
+  ProfilingContextMenuHeading,
+  ProfilingContextMenuItemButton,
+} from 'sentry/components/profiling/profilingContextMenu';
+import {NODE_ENV} from 'sentry/constants';
+import {IconChevron, IconCopy} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useContextMenu} from 'sentry/utils/profiling/hooks/useContextMenu';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useUser} from 'sentry/utils/useUser';
+
+type TraceElement = HTMLElement | SVGElement;
+
+const CURSOR_OFFSET_RIGHT = 4;
+const CURSOR_OFFSET_LEFT = 8;
+const CURSOR_OFFSET_TOP = 8;
+const CURSOR_OFFSET_BOTTOM = 4;
+
+export function SentryComponentInspector({theme}: {theme: Theme}) {
+  const tooltipRef = useRef<HTMLDivElement>(null);
+  const contextMenuElementRef = useRef<HTMLDivElement>(null);
+  const skipShowingTooltipRef = useRef<boolean>(false);
+
+  const user = useUser();
+  const organization = useOrganization({allowNull: true});
+  const [state, setState] = useState<{
+    enabled: null | 'inspector' | 'context-menu';
+    trace: TraceElement[] | null;
+  }>({
+    enabled: null,
+    trace: [],
+  });
+
+  const contextMenu = useContextMenu({container: tooltipRef.current});
+  const [contextMenuTrace, setContextMenuTrace] = useState<TraceElement[] | null>(null);
+
+  const tooltipPositionRef = useRef<{
+    left: number;
+    mouse: {x: number; y: number};
+    top: number;
+  } | null>(null);
+
+  const contextMenuPositionRef = useRef<{
+    left: number;
+    top: number;
+  } | null>(null);
+
+  const copyToClipboard = useCallback((text: string) => {
```

Comment:
we have a useClipboard or something for this btw

Reply:
https://github.com/getsentry/sentry/blob/master/static/app/utils/useCopyToClipboard.tsx

Reply:
I might do that yeah, just not in this PR